datalad/core/distributed/push.py

```diff
@@ -218,18 +218,17 @@ def __call__(
                     else 'No targets configured in dataset.'))
             return
 
-        if since:
-            # will blow with ValueError if unusable
-            ds_repo.get_hexsha(since)
-
-        if not since and since is not None:
-            # special case: --since=''
+        if since == '^':
             # figure out state of remote branch and set `since`
             since = _get_corresponding_remote_state(ds_repo, to)
             if not since:
                 lgr.info(
                     "No tracked remote for active branch, "
                     "detection of last pushed state not in effect.")
+        elif since:
+            # will blow with ValueError if unusable
+            ds_repo.get_hexsha(since)
```

Comment:
Thanks to the comment this suggestion
```suggestion
            since = ds_repo.get_hexsha(since)
```
is probably (who knows what helper functions do/report) is necessary. Just a note to myself then

Reply:
This would turn a potentially symbolic ref into a hexsha and make subsequent log messages harder to decode. I'd rather not do that.